lib/mongo_ecto/connection.ex

```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
       fun.()
     rescue
       e in Mongo.Error ->
-        stacktrace = System.stacktrace
         case e do
           %Mongo.Error{code: 11000, message: msg} ->
             {:invalid, constraint(msg)}
           other ->
+            stacktrace = System.stacktrace
```

Comment:
Why is this moved? It's better to acquire the stacktrace for reraising at the beginning of error handling.